File: pubnub-java/src/main/java/com/pubnub/api/endpoints/channel_groups/ListAllChannelGroup.java
package com.pubnub.api.endpoints.channel_groups;

import com.pubnub.api.models.consumer.channel_group.PNChannelGroupsListAllResult;
import com.pubnub.internal.InternalPubNubClient;
import com.pubnub.internal.endpoints.DelegatingEndpoint;
import lombok.experimental.Accessors;

@Accessors(chain = true, fluent = true)
public class ListAllChannelGroup extends DelegatingEndpoint<PNChannelGroupsListAllResult> {

    public ListAllChannelGroup(InternalPubNubClient pubnub) {
        super(pubnub);
    }

    @Override
    protected com.pubnub.internal.Endpoint<?, PNChannelGroupsListAllResult> createAction() {
        return pubnub.listAllChannelGroups();
    }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/channel_groups/ListAllChannelGroup.kt
package com.pubnub.api.endpoints.channel_groups

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.channel_group.PNChannelGroupsListAllResult

/**
 * @see [PubNub.listAllChannelGroups]
 */
actual interface ListAllChannelGroup : Endpoint<PNChannelGroupsListAllResult>
